### PR TITLE
feat: Cognite3DModel.getSubtreeIndices() now returns a range

### PR DESCRIPTION
**Pull request by larsmoa**
Also update migration doc to include how to migration use of `applyToChildren=true`.

### CODE REVIEW

File: documentation/docs/migration-guide.md [hunk 1]
```diff
@@ -111,6 +111,31 @@ This example could easily have been migrated using `ByNodePropertyNodeSet`, but
...
+When this was `true`, Reveal would apply the styling to all descendants. When migrating
+to the new version, adding the root tree index of the subtree to `ByTreeIndexNodeSet` will not have
+the same result. Instead, use `Cognite3DModel.getSubtreeIndices()` to get the full range of tree
+indices.
+
+```js
+// In Reveal 1.x:
+// model.hideNodeByTreeIndex(subtreeRootTreeIndex, true);
+
+// Wrong - will only affect a single tree index, not all descendants
```
Comment (maksnester) [inline documentation/docs/migration-guide.md:126]:
```suggestion
// ❌ Wrong - will only affect a single tree index, not all descendants
```

File: documentation/docs/migration-guide.md [hunk 2]
```diff
@@ -111,6 +111,31 @@ This example could easily have been migrated using `ByNodePropertyNodeSet`, but
...
+const indexSet = new IndexSet();
+indexSet.add(subtreeRootIndex);
+const nodeSet = new ByTreeIndexNodeSet(indexSet);
+model.addStyledSet(nodeSet, { visible: false });
+
+// Correct - all descendants are affected
```
Comment (maksnester) [inline documentation/docs/migration-guide.md:132]:
```suggestion
// ✅ Correct - all descendants are affected
```

File: documentation/docs/migration-guide.md [hunk 3]
```diff
@@ -111,6 +111,31 @@ This example could easily have been migrated using `ByNodePropertyNodeSet`, but
 other cases this might not be as straight forward as shown.
 :::
 
+#### A note on `applyToAllChildren` to style all descendants of a node
+
+In previous versions, all styling functions accepted an optional `applyToAllChildren`-argument.
+When this was `true`, Reveal would apply the styling to all descendants. When migrating
+to the new version, adding the root tree index of the subtree to `ByTreeIndexNodeSet` will not have
+the same result. Instead, use `Cognite3DModel.getSubtreeIndices()` to get the full range of tree
```
Comment (maksnester) [inline documentation/docs/migration-guide.md:119]: should subtreeSize be mentioned here as a valid alternative?

**Review by maksnester — APPROVED** [review]
LGTM but suggested emoji for do/don't section - they are more eye-catchy

File: documentation/docs/migration-guide.md (hunk 3 above)
Comment (larsmoa) [reply documentation/docs/migration-guide.md]: Good point - added

**Pull request merged**